panther_hardware_interfaces/include/panther_hardware_interfaces/gpio_controller.hpp

```diff
@@ -0,0 +1,294 @@
+   * @return 'true' if the motor control pin value is successfully set, 'false' otherwise.
+   */
+  bool FanEnable(const bool enable) override;
+
+  /**
+   * @brief Controls AUX power source based on the 'enable' parameter.
+   *
+   * @param enable Set to 'true' to enable the motors, 'false' to disable.
+   * @return 'true' if the motor control pin value is successfully set, 'false' otherwise.
+   */
+  bool AUXEnable(const bool enable) override;
+
+  /**
+   * @brief Controls the digital power source based on the 'enable' parameter.
+   *
+   * @param enable Set to 'true' to enable the motors, 'false' to disable.
+   * @return 'true' if the motor control pin value is successfully set, 'false' otherwise.
+   */
+  bool VDIGEnable(const bool enable) override;
+
+  bool ChargerEnable(const bool enable) override;
+
+private:
+  /**
+   * @brief Vector containing GPIO pin configuration information such as pin direction, value, etc.
+   */
+  const std::vector<panther_gpiod::GPIOInfo> gpio_config_info_storage_{
+    panther_gpiod::GPIOInfo{panther_gpiod::GPIOPin::WATCHDOG, gpiod::line::direction::OUTPUT},
+    panther_gpiod::GPIOInfo{panther_gpiod::GPIOPin::AUX_PW_EN, gpiod::line::direction::OUTPUT},
+    panther_gpiod::GPIOInfo{panther_gpiod::GPIOPin::CHRG_DISABLE, gpiod::line::direction::OUTPUT},
+    panther_gpiod::GPIOInfo{panther_gpiod::GPIOPin::DRIVER_EN, gpiod::line::direction::OUTPUT},
+    panther_gpiod::GPIOInfo{panther_gpiod::GPIOPin::E_STOP_RESET, gpiod::line::direction::INPUT},
+    panther_gpiod::GPIOInfo{panther_gpiod::GPIOPin::FAN_SW, gpiod::line::direction::OUTPUT},
+    panther_gpiod::GPIOInfo{panther_gpiod::GPIOPin::GPOUT1, gpiod::line::direction::OUTPUT},
+    panther_gpiod::GPIOInfo{panther_gpiod::GPIOPin::GPOUT2, gpiod::line::direction::OUTPUT},
+    panther_gpiod::GPIOInfo{panther_gpiod::GPIOPin::GPIN1, gpiod::line::direction::INPUT},
+    panther_gpiod::GPIOInfo{panther_gpiod::GPIOPin::GPIN2, gpiod::line::direction::INPUT},
+    panther_gpiod::GPIOInfo{panther_gpiod::GPIOPin::SHDN_INIT, gpiod::line::direction::INPUT},
+    panther_gpiod::GPIOInfo{panther_gpiod::GPIOPin::VDIG_OFF, gpiod::line::direction::OUTPUT},
+    panther_gpiod::GPIOInfo{panther_gpiod::GPIOPin::VMOT_ON, gpiod::line::direction::OUTPUT},
+    panther_gpiod::GPIOInfo{
+      panther_gpiod::GPIOPin::CHRG_SENSE, gpiod::line::direction::INPUT, true},
+  };
+  std::unique_ptr<Watchdog> watchdog_;
+};
+
+class GPIOControllerPTH10X : public GPIOControllerInterface
+{
+public:
+  /**
+   * @brief Initializes the GPIODriver and powers on the motors.
+   */
+  void Start() override;
+
+  /**
+   * @brief Placeholder method indicating lack of hardware E-Stop support for the robot in this
+   * version.
+   *
+   * @return Always returns true.
+   */
+  bool EStopTrigger() override;
+
+  /**
+   * @brief Checks if the motors are powered up (when STAGE2 is active) without controlling any
+   * GPIO.
```

Comment:
```suggestion
   * @brief Checks if the motors are powered up (when STAGE2_INPUT is active) without controlling any
   * GPIO.
```
or STAGE2_INPUT/main switch is set to STAGE2 position